diff --git a/build/tasks/kernel.mk b/build/tasks/kernel.mk
--- a/build/tasks/kernel.mk
+++ b/build/tasks/kernel.mk
@@ -462,7 +462,7 @@
 KERNEL_MODULES_OUT := $(TARGET_OUT_VENDOR_DLKM)
 KERNEL_DEPMOD_STAGING_DIR := $(KERNEL_BUILD_OUT_PREFIX)$(call intermediates-dir-for,PACKAGING,depmod_vendor)
 KERNEL_MODULE_MOUNTPOINT := vendor_dlkm
-KERNEL_MODULES_PARTITION_FILE_LIST := $(vendor_dlkmimage_intermediates)/file_list.txt
+KERNEL_MODULES_PARTITION_FILE_LIST := $(if $(BUILDING_VENDOR_DLKM_IMAGE),$(vendor_dlkmimage_intermediates)/file_list.txt)
 $(INSTALLED_VENDOR_DLKMIMAGE_TARGET): $(TARGET_PREBUILT_INT_KERNEL)
 else
 KERNEL_MODULES_OUT := $(TARGET_OUT_VENDOR)
@@ -479,7 +479,7 @@
 SYSTEM_KERNEL_MODULES_OUT := $(TARGET_OUT_SYSTEM_DLKM)
 SYSTEM_KERNEL_DEPMOD_STAGING_DIR := $(KERNEL_BUILD_OUT_PREFIX)$(call intermediates-dir-for,PACKAGING,depmod_system_dlkm)
 SYSTEM_KERNEL_MODULE_MOUNTPOINT := system_dlkm
-SYSTEM_KERNEL_MODULES_PARTITION_FILE_LIST := $(system_dlkmimage_intermediates)/file_list.txt
+SYSTEM_KERNEL_MODULES_PARTITION_FILE_LIST := $(if $(BUILDING_SYSTEM_DLKM_IMAGE),$(system_dlkmimage_intermediates)/file_list.txt)
 $(INSTALLED_SYSTEM_DLKMIMAGE_TARGET): $(TARGET_PREBUILT_INT_KERNEL)
 else ifneq ($(SYSTEM_KERNEL_MODULES)$(filter $(TARGET_KERNEL_MIXED_MODE),true),)
 SYSTEM_KERNEL_MODULES_OUT := $(TARGET_OUT)
